src/notebooks/controllers/controllerRegistration.ts

```diff
@@ -43,12 +48,16 @@ export class ControllerRegistration implements IControllerRegistration {
     }
     private registeredControllers = new Map<string, VSCodeNotebookController>();
     private creationEmitter = new EventEmitter<IVSCodeNotebookController>();
+    private changeEmitter = new EventEmitter<IVSCodeNotebookControllerUpdateEvent>();
     private registeredMetadatas = new Map<string, KernelConnectionMetadata>();
     private inKernelExperiment = false;
 
     public get onCreated(): Event<IVSCodeNotebookController> {
         return this.creationEmitter.event;
```

Comment:
can you remove the onCreated event, so we'd have a consistent way to look for changes

Reply:
@DonJayamanne I like the idea of removing `onCreated` event and move towards `onChange`. Currently `onCreated` is being used in half a dozen places, removing it would require changes to them and I'm not inclined to replacing `onCreated(handler)` to `onChange(e => e.added.foreach(controller => handler(controller)` naively, but rather optimizing how we handle batch update events in each component. Does this make better sense? I can also remove them simply if that's more preferred.

Reply:
Since I'm working in this space, I'll clean this up

Reply:
@DonJayamanne ❤️ thanks, I created https://github.com/microsoft/vscode-jupyter/issues/11694 to track this. I can help with the controllers part too.